src/primitives/compression.h

```diff
@@ -0,0 +1,475 @@
+#ifndef BITCOIN_PRIMITIVES_COMPRESSION_H
+#define BITCOIN_PRIMITIVES_COMPRESSION_H
+
+#include <streams.h>
+#include <secp256k1.h>
+#include <secp256k1_recovery.h>
+#include <secp256k1_extrakeys.h>
+#include <secp256k1_schnorrsig.h>
+#include <consensus/amount.h>
+#include <script/script.h>
+#include <serialize.h>
+#include <uint256.h>
+#include <primitives/transaction.h>
+#include <script/standard.h>
+#include <cmath>
+#include <logging.h>
+#include <util/strencodings.h>
+#include <bitset>
+
+class CCompressedTxId
+{
+private:
+    uint32_t m_block_height;
+    uint32_t m_block_index;
+    uint256 m_txid;
+public:
+    const uint32_t&  block_height() const  { return m_block_height; }
+    const uint32_t& block_index() const { return m_block_index; }
+    const uint256& txid() const { return m_txid; }
+
+    explicit CCompressedTxId(const uint32_t& block_height, const uint32_t& block_index);
+    explicit CCompressedTxId(const uint256& txid);
+
+    bool IsCompressed() const {
+        return m_txid.IsNull();
+    }
+
+    friend bool operator==(const CCompressedTxId& a, const CCompressedTxId& b) {
+        return a.m_block_height == b.m_block_height && a.m_block_index == b.m_block_index && a.m_txid    == b.m_txid;
```

Comment:
In 89b52782f8d6aca75d49048da1e137fbf615c0e6

nit: weird spacing

Reply:
Solved, Thanks!